site/content/blog/2020-09-07-improving-performance-through-load-testing.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "Improving performance (and more) through load testing"
+slug: improving-performance-through-load-testing
+date: 2020-09-07
+categories:
+    - "performance"
+    - "testing"
+author: Claudio Costa
+github: streamer45
+community: claudio.costa
+canonicalUrl: https://mattermost.com/blog/improving-performance-through-load-testing/
+---
+
+Have you ever wondered how many active users your application can handle at the same time? If so, you're not alone. Here at Mattermost we're building a highly concurrent messaging platform for team collaboration that needs to potentially serve up to several thousands of users simultaneously. 
+
+While standard functional testing (e.g. unit tests) is critical to verify correct behavior of your application, it's usually not sufficient to guarantee its performance at scale. This is where some form of performance testing can be particularly useful. Benchmarks do a good job at measuring the performance impact of single functions or limited portions of a codebase, but they can't capture how an application is going to perform as a whole under real-world conditions. To achieve this you need **load testing**.
+
+## What is load testing?
+
+Load testing is the process of applying a certain amount of load to a target system in order to understand how it behaves.
+The more realistic the generated load is, the more accurate and insightful the final results will be.
+
+## Why does it matter?
```

Comment:
Maybe this could be a good point to expand on in a possible follow up article which will be more technical and completely focused on our solution. Here my plan is more to do a gentle introduction on the value of load-testing as a general technique and mention how we solved the problem but without going into much depth.

Reply:
I see, makes sense :)